backend/lib/use-cases/case-assignment/case-assignment.ts

```diff
@@ -71,14 +77,53 @@ export class CaseAssignmentUseCase {
     const assignmentRepo = Factory.getAssignmentRepository(context);
     context.logger.info(MODULE_NAME, 'New assignments:', newAssignments);
 
+    const bCase = await casesRepo.getSyncedCase(caseId);
+    const officesGateway = getOfficesGateway(context);
+    const offices = await officesGateway.getOffices(context);
+    const divisionCodeMap = mapDivisionCodeToUstpOffice(offices);
+    const officeCode = divisionCodeMap.get(bCase.caseId.substring(0, 3)).officeCode;
+
+    const officesRepo = getOfficesRepository(context);
+    const calls = [];
+    const validatedAssignments: CamsUserReference[] = [];
+    newAssignments.forEach((assignment) => {
+      calls.push(
+        officesRepo
+          .search({ officeCode, userId: assignment.id, role })
+          .then((response) => {
+            return response.find((staff) => {
+              if (staff.roles.includes(role as CamsRole) && staff.name === assignment.name) {
+                return true;
+              }
+            });
+          })
+          .catch((_err) => {
+            // Do something?
```

Comment:
We should discuss if there is any need to have a catch either here or in the `Promise.all()` lower down.